cli/tests/test_global_opts.rs

```diff
@@ -304,16 +304,16 @@ fn test_invalid_filesets_looking_like_filepaths() {
     let repo_path = test_env.env_root().join("repo");
 
     let stderr = test_env.jj_cmd_failure(&repo_path, &["file", "show", "abc~"]);
-    insta::assert_snapshot!(stderr, @r#"
+    insta::assert_snapshot!(stderr, @r"
     Error: Failed to parse fileset: Syntax error
     Caused by:  --> 1:5
       |
     1 | abc~
       |     ^---
       |
       = expected `~` or <primary>
-    Hint: See https://jj-vcs.github.io/jj/latest/filesets/ for filesets syntax, or for how to match file paths.
-    "#);
+    Hint: See https://jj-vcs.github.io/jj/latest/filesets/ for filesets syntax. Names with special symbols must be quoted.
```

Comment:
I don't think this change will help because the user might not know why literal path doesn't work. If they knew, they can easily parse the error message.

Maybe we can instead print an escaped version of user-provided literal path. Since quoted syntax may be shell-dependent, it would be easier if we had `literal:` prefix or something.
https://github.com/jj-vcs/jj/issues/4509#issuecomment-2465963426